adds release note for terminus 4.0.0-alpha1 (#9472)
pantheon-systems/documentation: source/releasenotes/2025-03-07-terminus-4-0-0-alpha1.md

---
title: "Terminus 4.0.0-alpha1 release now available"
published_date: "2025-03-07"
categories: [tools-apis]
---

> A Terminus [4.0.0-alpha1](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1) _pre-release_ is now available as of March 07, 2025. The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.2 and below.

Reviewer: I don't think we should repeat the date from the metadata of the release note, especially since `4.0.0-alpha1` was tagged on March 6 (though "available" is fuzzier if there were no obvious install/upgrade instructions)

For more information about this release, visit the [GitHub release page](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1).

## PHP support for older PHP versions in Terminus 3.x

Terminus 3.x is **not compatible with PHP 8.4**. The recommended PHP version to use locally with Terminus 3.x is 8.3, which can be installed via Homebrew (on MacOS) by specifying the version when installing the `php` package, e.g. `brew install php@8.3`. For more information see [the `php` Homebrew formula](https://formulae.brew.sh/formula/php@8.3).

## How to upgrade to Terminus 4.x while in alpha

> During the pre-release development phase of Terminus 4, the only way to install Terminus is [directly from the `phar`](https://docs.pantheon.io/terminus/install#windows-and-linux). If you manage your installation via Homebrew on macOS, you will not be able to update to Terminus 4 until it is generally available and the Brew formula is updated.

Reviewer: This strikes me as a shortcoming of however we're administering `brew`. If it is possible to specify different versions of PHP via brew, why not different versions of Terminus?

Reviewer: Or maybe I'm just not close enough to Brew processes. As I look at closer at PHP, yes it is possible to ask for a specific older version like `php@8.3` but that does not replace (at least in my short testing) the default version of PHP 8.4 installed when no version is specified. And the next version of PHP 8.5 is just not available

Author: I don't think this is a limitation of how we're using `brew`. I think this is a limitation because 4.0.0-alpha1 is a _pre-release_ (and therefore not a real _published package_). If we want to keep the alpha, pre-release state, I think we need to install via `phar`.

---
title: "Terminus 4.0.0-alpha1 release now available"
published_date: "2025-03-07"
categories: [tools-apis]
---

A Terminus [4.0.0-alpha1](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1) _pre-release_ is now available.

> The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.2 and below.

Author: ```suggestion
The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.
```

For more information about this release, visit the [GitHub release page](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1).

> ## PHP support for older PHP versions in Terminus 3.x

Author: ```suggestion
## Support for older PHP versions in Terminus 3.x
```

> Terminus 3.x is **not compatible with PHP 8.4**. The recommended PHP version to use locally with Terminus 3.x is 8.3, which can be installed via Homebrew (on MacOS) by specifying the version when installing the `php` package, e.g. `brew install php@8.3`. For more information see [the `php` Homebrew formula](https://formulae.brew.sh/formula/php@8.3).

Author: ```suggestion
Terminus 3.x is **not compatible with PHP 8.4**. The recommended PHP version to use locally with Terminus 3.x is 8.3. 

On MacOS with Homebrew PHP 8.3 can be installed by specifying the version when installing the `php` package, e.g. `brew install php@8.3`. If you already have PHP installed at the default version (8.4), you may need to use [`brew unlink php`](https://docs.brew.sh/Manpage#unlink---dry-run-installed_formula-) and [`brew link php@8.3`](https://docs.brew.sh/Manpage#link-ln-options-installed_formula-) to use the corrected version.

For more information see [the `php` Homebrew formula](https://formulae.brew.sh/formula/php@8.3).
```

---
title: "Terminus 4.0.0-alpha1 release now available"
published_date: "2025-03-07"
categories: [tools-apis]
---

A Terminus [4.0.0-alpha1](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1) _pre-release_ is now available.
The next generation of Terminus adds support for PHP 8.4 and removes support for PHP versions 8.1 and below.

For more information about this release, visit the [GitHub release page](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0-alpha1).

## Support for older PHP versions in Terminus 3.x

Terminus 3.x is **not compatible with PHP 8.4**. The recommended PHP version to use locally with Terminus 3.x is 8.3. 

> On MacOS with Homebrew PHP 8.3 can be installed by specifying the version when installing the `php` package, e.g. `brew install php@8.3`. If you already have PHP installed at the default version (8.4), you may need to use [`brew unlink php`](https://docs.brew.sh/Manpage#unlink---dry-run-installed_formula-) and [`brew link php@8.3`](https://docs.brew.sh/Manpage#link-ln-options-installed_formula-) to use the corrected version.

Author: ```suggestion
On macOS with Homebrew PHP 8.3 can be installed by specifying the version when installing the `php` package, e.g. `brew install php@8.3`. If you already have PHP installed at the default version (8.4), you may need to use [`brew unlink php`](https://docs.brew.sh/Manpage#unlink---dry-run-installed_formula-) and [`brew link php@8.3`](https://docs.brew.sh/Manpage#link-ln-options-installed_formula-) to use the corrected version.
```

For more information see [the `php` Homebrew formula](https://formulae.brew.sh/formula/php@8.3).

## How to upgrade to Terminus 4.x while in alpha
During the pre-release development phase of Terminus 4, the only way to install Terminus is [directly from the `phar`](https://docs.pantheon.io/terminus/install#windows-and-linux). If you manage your installation via Homebrew on macOS, you will not be able to update to Terminus 4 until it is generally available and the Brew formula is updated.

```shell{promptUser: user}
mkdir -p ~/terminus && cd ~/terminus
curl -L https://github.com/pantheon-systems/terminus/releases/download/4.0.0-alpha1/terminus.phar --output terminus
chmod +x terminus
./terminus self:update
sudo ln -s ~/terminus/terminus /usr/local/bin/terminus
```

> If you have previously installed Terminus using Brew, you will need to run `brew unlink terminus` to remove the symbollic links that Homebrew uses for Terminus.

Reviewer: ```suggestion
If you have previously installed Terminus using Brew, you will need to run `brew unlink terminus` to remove the symbollic links that Homebrew uses for Terminus.

If you have questions or concerns around Terminus 4, please use the [Terminus issue queue](https://github.com/pantheon-systems/terminus).
```
